src/ethereum_provider.js

```diff
@@ -263,7 +263,10 @@ class TrustWeb3Provider extends BaseProvider {
     var message;
     let address;
 
-    if (this.address === payload.params[0].toLowerCase()) {
+    if (
+      typeof payload.params[0].toLowerCase() === "string" &&
```

Comment:
if `payload.params[0]` is not a string (eg: `undefined` or `null`), `.toLowerCase()` would theoretically fail 🤔

I guess it should be `typeof payload.params?.[0] === "string" && ...`

Reply:
same to the below verification.

Reply:
In theory should not but happy to add more defensive code 💪